app/src/main/java/com/example/authapp/ui/SignOutUser.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.authapp.ui
+
+import android.app.Activity
+import android.content.Intent
+import android.widget.Toast
+import com.example.authapp.R
+import com.example.authapp.handler.PreferenceHandler
+import com.example.authapp.newtorkOperation.FirebaseAuthOperation
+import com.example.authapp.repository.UserRepository
+import com.example.authapp.usecase.SignOutUserUseCase
+import com.example.authapp.usecase.SignOutUserUseCaseImpl
+
+class SignOutUser {
```

Comment:
instead of making a companion object you could have make the entire class static.